fenix/app/src/main/java/org/mozilla/fenix/settings/search/SearchEngineFragment.kt

```diff
@@ -32,6 +32,12 @@ class SearchEngineFragment : PreferenceFragmentCompat() {
             if (unifiedSearchUI) R.xml.search_settings_preferences else R.xml.search_preferences,
             rootKey,
         )
+
+        // Visibility should be set before the view has been created, to avoid visual glitches.
+        requirePreference<SwitchPreference>(R.string.pref_key_show_search_engine_shortcuts).apply {
+            isVisible = !context.settings().showUnifiedSearchFeature
+        }
```

Comment:
We want to keep the option of turning it off in case it affects search revenue. So unless we are ready to remove all non unified search code, we should support both behaviours. The estimate for the 
"probation" period would be around two release, from what I have heard.

If we disable it in XML, then we will get the glitch again if `showUnifiedSearchFeature == false`. 
I would set the visibility before the view is created, and then remove the `SwitchPreference` completely once `UnifiedSearch` is verified not to be a regression.